pkg/volume/empty_dir/quota.go

```diff
@@ -0,0 +1,128 @@
+package empty_dir
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// An object for applying quota to an empty dir volume.
+type quotaApplicator interface {
+	// Apply the quota to the given EmptyDir path:
+	Apply(dir string, medium api.StorageMedium, pod *api.Pod) error
+}
+
+type xfsQuotaApplicator struct {
+}
+
+func (xqa *xfsQuotaApplicator) Apply(dir string, medium api.StorageMedium, pod *api.Pod, fsGroup *int64, quota int) error {
+	if medium == api.StorageMediumMemory {
+		glog.V(3).Infof("skipping quota application due to memory storage medium.")
+		return nil
+	}
+	if !xqa.isXfs(dir) {
+		glog.V(3).Infof("%s is not on an XFS filesystem, skipping quota application.")
+		return nil
+	}
+	if fsGroup == nil {
+		// This indicates the operation matched an SCC with FSGroup strategy RunAsAny.
+		// Not an error condition.
+		glog.V(3).Infof("Unable to apply XFS quota, no FSGroup specified.")
+		return nil
+	}
+
+	volDevice, err := getFsDevice(dir)
+	if err != nil {
+		return err
+	}
+
+	err = xqa.applyQuota(volDevice, quota, *fsGroup)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Check if the empty dir is on an XFS filesystem:
+func (xqa *xfsQuotaApplicator) isXfs(dir string) bool {
```

Comment:
I prefer to return (bool, error) so the caller can distinguish "yes" vs. "no" vs. "I don't know because of an error", but don't feel too strongly.